src/form-host/form-wireup.js

```diff
@@ -0,0 +1,57 @@
+class FormWireup {
+  constructor(formHtml, formModel, formXml, userSettingsDoc, contactSummary, formName) {
+    const formWrapper = this._getFormWrapper();
+    formWrapper.user = userSettingsDoc;
+    if (contactSummary) {
+      formWrapper.contactSummary = contactSummary.context;
+    }
+    formWrapper.formId = formName;
+    formWrapper.formHtml = formHtml;
+    formWrapper.formModel = formModel;
+    formWrapper.formXml = formXml;
+  }
+
+  render(content) {
+    return new Promise((resolve) => {
+      const formWrapper = this._getFormWrapper();
+      formWrapper.addEventListener('onRender', () => resolve());
+      formWrapper.content = content;
+    });
+  }
+
+  async save() {
+    return new Promise((resolve, reject) => {
+      const observer = new MutationObserver(() => {
```

Comment:
mutationobservers really coming in handy. are they global DOM changes? should they be scoped at all - dunno if embedding form-wireup.js into something like explorer could trigger these events erroneously or something?

no specific scenario in mind

Reply:
Very handy, but it does kind of feel like a hack that demonstrates some missing events that cht-form should be pushing.... 

> are they global DOM changes? should they be scoped at all

I am not an expert at all on MutationObservers, but I tried to get these scoped as tight as possible.  Below, where we call `observe`, you can see that it is only going to be looking at the DOM elements _within the `#enketo-wrapper`, and out of those it will only fire for changes to the `status` attribute.

> dunno if embedding form-wireup.js into something like explorer could trigger these events erroneously or something?

I don't think so. (Though once again, I am really only pretending to know anything about web development....)  We should only register the observer if this `save` function is called.  Which, if something like the Project Explorer is calling `save`, then I guess it probably needs the observer functionality to work.  Otherwise, as long as the Project Explorer is not using `save`, then this observer should not affect anything...